### PR TITLE
Switch model templates to FP16

### PR DESCRIPTION
**Pull request by sovrasov**
Note: now precision for aux models is taken from the main mode's config. NNCF task operates in FP32.

### CODE REVIEW

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Please, provide a link to the test results to confirm that optimization flow NNCF/POT still works.

**Review by alexsu52 — COMMENTED** [review]
You should add to the end of the `__init__` method of the NNCF task the following code to override `mix_precision` attribute in the model:  
```
        # Set model attributes
        set_model_attr(self._model, 'mix_precision', self._cfg.train.mix_precision)
```

**Review by alexsu52 — APPROVED** [review]

**Review by alexsu52 — CHANGES_REQUESTED** [review]
ref #77169

**Review by kprokofi — APPROVED** [review]

**Review by alexsu52 — COMMENTED** [review]
NNCF was updated and supports FP16 training (https://github.com/openvinotoolkit/deep-object-reid/pull/175). Please rebase and run tests.

File: torchreid/utils/tools.py [hunk 1]
```diff
@@ -168,6 +168,14 @@ def get_model_attr(model, attr):
     return getattr(model, attr)
 
 
+def set_model_attr(model, attr, value):
```
Comment (alexsu52) [inline torchreid/utils/tools.py:171]: Please, use set_model_attr everywhere, for example there: https://github.com/openvinotoolkit/deep-object-reid/blob/481396ce0dcb009b04c6232ad1d982d66a9a2858/torchreid/integration/sc/inference_task.py#L280

**Review by alexsu52 — COMMENTED** [review]
LGTM

**Comment by sovrasov** [comment]
ote-test/967, ote-test/966

File: torchreid/utils/tools.py (hunk 1 above)
Comment (sovrasov) [reply torchreid/utils/tools.py:171]: Will fix that in another PR.

**Pull request merged**